pants.toml

```diff
@@ -22,3 +23,6 @@ enabled = false
 
 [environments-preview.names]
 workspace = "//:workspace"
+
+[python]
+interpreter_constraints = ">=3.9"
```

Comment:
We encourage others to pin ICs to a single version, so we should probably model that here. Why is python involved at all though?

Reply:
Bats me. Pants errors if that option is not set in this repository:

```
% pants run //:project_image
09:45:01.26 [INFO] waiting for pantsd to start...
09:45:01.37 [INFO] pantsd started
09:45:01.70 [INFO] Initializing scheduler...
09:45:01.77 [INFO] Initializing Nailgun pool for 32 processes...
09:45:01.97 [INFO] Scheduler initialized.
09:45:02.88 [ERROR] 1 Exception encountered:

Engine traceback:
  in `run` goal

OptionsError: You must explicitly specify the default Python interpreter versions your code is intended to run against.

You specify these interpreter constraints using the `interpreter_constraints` option in the `[python]` section of pants.toml.

We recommend constraining to a single interpreter minor version if you can, e.g., `interpreter_constraints = ['==3.11.*']`, or at least a small number of interpreter minor versions, e.g., `interpreter_constraints = ['>=3.10,<3.12']`.

Individual targets can override these default interpreter constraints, if different parts of your codebase run against different python interpreter versions in a single repo.

See https://www.pantsbuild.org/2.23/docs/python/overview/interpreter-compatibility for details.
```

Reply:
Also, this is an example repository, I'd rather not pin Python to a specific version which I have but which users of this example may not have.

Reply:
I am going to add a comment explaining the issue. This will allow me to land the PR so I can get the blog post out. We can fix up ICs afterward as a follow-on.

Reply:
Comment added.